src/ngx_http_auth_jwt_module.c

```diff
@@ -670,6 +672,55 @@ static char *get_jwt(ngx_http_request_t *r, ngx_str_t jwt_location)
       jwtPtr = ngx_str_t_to_char_ptr(r->pool, jwtCookieVal);
     }
   }
+  else if (jwt_location.len > sizeof(QUERY_STRING_PREFIX) && ngx_strncmp(jwt_location.data, QUERY_STRING_PREFIX, sizeof(QUERY_STRING_PREFIX) - 1) == 0) {
+    jwt_location.data += sizeof(QUERY_STRING_PREFIX) - 1;
+    jwt_location.len -= sizeof(QUERY_STRING_PREFIX) - 1;
 
+    size_t token_key_start = 0, token_value_start = 0, token_end = 0;
+
+    bool found_token = search_token_from_args(
+      &jwt_location,
+      &r->args,
+      &token_key_start,
+      &token_value_start,
+      &token_end
+    );
+
+    if (!found_token) return NULL;
+
+    int token_len = token_end - token_value_start;
+
+    jwtPtr = ngx_palloc(r->pool, token_len + 1);
+    if (jwtPtr != NULL) {
+      ngx_memcpy(jwtPtr, r->args.data + token_value_start, token_len);
+      *(jwtPtr + token_len) = '\0';
+    }
+
+    // strip first or last & from args
+    if (token_key_start > 0) {
+      token_key_start--;
+    } else if (token_end < (r->args.len - 1)) {
+      token_end++;
+    }
+
+    size_t mutated_args_len = 0;
+
+    // Strip key from args
+    u_char *args_ptr = create_args_without_token(
+      r->pool,
+      &r->args,
+      token_key_start,
+      token_end,
+      &mutated_args_len
+    );
+
+    if (args_ptr == NULL) return jwtPtr;
+
+    int free_ok = ngx_pfree(r->pool, r->args.data);
+    if (free_ok == NGX_OK) {
+      r->args.data = args_ptr;
+      r->args.len = mutated_args_len;
+    }
```

Comment:
Can just return at the end. So here, just update the args.

```suggestion
    if (args_ptr != NULL)
    {
      int free_ok = ngx_pfree(r->pool, r->args.data);
      
      if (free_ok == NGX_OK) {
        r->args.data = args_ptr;
        r->args.len = mutated_args_len;
      }
    }
```

Reply:
Thanks, fixed in commit: https://github.com/TeslaGov/ngx-http-auth-jwt-module/pull/107/commits/7dd8b005b8e932175e7cef4f34c026eb1b9c6fdf